cluster/v1/types.go

```diff
@@ -82,6 +86,43 @@ type ClientConfig struct {
 	CABundle []byte `json:"caBundle,omitempty"`
 }
 
+// The managed cluster this Taint is attached to has the "effect" on
+// any placement that does not tolerate the Taint.
+type Taint struct {
+	// The taint key to be applied to a cluster. e.g. bar or foo.example.com/bar.
+	// The regex it matches is (dns1123SubdomainFmt/)?(qualifiedNameFmt)
+	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=`^([a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*/)?(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])$`
+	// +kubebuilder:validation:MaxLength=316
+	Key string `json:"key"`
+	// The taint value corresponding to the taint key.
+	// +optional
+	// +kubebuilder:validation:MaxLength=1024
+	Value string `json:"value,omitempty"`
+	// Required. The effect of the taint on placements that do not tolerate the taint.
+	// Valid effects are NoSelect, PreferNoSelect and NoSelectIfNew.
+	Effect TaintEffect `json:"effect"`
+	// TimeAdded represents the time at which the taint was added.
+	TimeAdded metav1.Time `json:"timeAdded"`
```

Comment:
should this be pointer?

Reply:
Since we always know the time the taint was added, David suggested we don't use a point for this field. See more details here:
https://github.com/open-cluster-management-io/enhancements/pull/22#discussion_r745047189

Reply:
add optional?

Reply:
If we make it optional, we may need a mutating webhook to fill this field automatically.

Reply:
I think you will always need it...

Reply:
> should this be pointer?

Why should this be a pointer or allow it to be optional if we always know the value?

Reply:
Users don't have to input `TimeAdded`, and the mutating webhook will set its value. So it's still required in the spec, but optional for user input.